consensus/src/lib.rs

```diff
@@ -173,13 +185,15 @@ impl Consensus {
 impl Consensus {
     async fn read_state(&self) -> Result<State, Error> {
         let raw_state = self.state_storage.read_file(STATE_FILE_NAME).await?;
-        let state: State = serde_spb::from_str(&raw_state)?;
+        let state: State = serde_spb::from_slice(&hex::decode(raw_state)?)?;
```

Comment:
Can you explain what is different between serde_spb and serde_json?

Reply:
serde_spb is a module for supporting en/decoding of data in Simeprby protocol both in binary and string. The format that the module actually uses is not the concern of the protocol. It just 'happens to' be using Json as its string format.